plansys2_core/include/plansys2_core/PlanSolverBase.hpp

```diff
@@ -52,7 +52,8 @@ class PlanSolverBase
   */
   virtual std::optional<plansys2_msgs::msg::Plan> getPlan(
     const std::string & domain, const std::string & problem,
-    const std::string & node_namespace = "") = 0;
+    const std::string & node_namespace = "",
+    const int solver_timeout = 15) = 0;
```

Comment:
Why `int` for the timeout? I would prefer a rclcpp::Duration? In the parameter is ok. You can document that this needs to be seconds, but internally I prefer this type.

```
using namespace std::chrono_literals;
...
const rclcpp::Duration  solver_timeout = 15s) = 0;
```